libportability-gfx/src/impls.rs

```diff
@@ -3126,25 +3120,55 @@ pub extern "C" fn gfxCmdFillBuffer(
 }
 #[inline]
 pub extern "C" fn gfxCmdClearColorImage(
-    _commandBuffer: VkCommandBuffer,
-    _image: VkImage,
-    _imageLayout: VkImageLayout,
-    _pColor: *const VkClearColorValue,
-    _rangeCount: u32,
-    _pRanges: *const VkImageSubresourceRange,
+    mut commandBuffer: VkCommandBuffer,
+    image: VkImage,
+    imageLayout: VkImageLayout,
+    pColor: *const VkClearColorValue,
+    rangeCount: u32,
+    pRanges: *const VkImageSubresourceRange,
 ) {
-    unimplemented!()
+    let subresource_ranges = unsafe {
+        slice::from_raw_parts(pRanges, rangeCount as _)
+    };
+    commandBuffer.clear_image(
+        match *image {
+            Image::Image(ref image) => image,
+            Image::Unbound(_) => panic!("Bound image expected!"),
+        },
+        conv::map_image_layout(imageLayout),
+        unsafe { mem::transmute(*pColor) },
+        unsafe { mem::zeroed() },
```

Comment:
Callers having to use `zeroed` here makes me wonder whether the API should simply accept `Option` (or we provide another way to make this cleaner)

Reply:
would probably be easier to implement `Default` on those